src/router/PrefixMatching/PrefixMap.php

```diff
@@ -19,11 +19,16 @@
 use namespace HH\Lib\{C, Dict, Keyset, Str, Vec};
 
 final class PrefixMap<T> {
+  private int $prefixLength;
+
   public function __construct(
     private dict<string, T> $literals,
     private dict<string, PrefixMap<T>> $prefixes,
     private dict<string, PrefixMapOrResponder<T>> $regexps,
   ) {
+    $this->prefixLength = C\is_empty($prefixes)
```

Comment:
This can be further improved by making prefix length another parameter, as it's already been computed  in ::groupByCommonPrefix

Reply:
wasn't sure which to use, inout or tuple return, inout seemed a bit faster locally, but not sure.